### ISSUE TITLE
Share at Timestamp UI Behaviour Changes

### ISSUE THREAD

**Issue by jas-kas** [getsentry/sentry#46796]
### Context

The context for this modal is:
- In the base case there's no `?t=` param expected to be in the browser url bar.
- You can actually get `?t=` or `?event_t=` in the browser url bar if you come from pages like Issue Details
  - Either of those params will not update as you play through the video, or click on a console log (for example). So if you copy+paste the url from the browser url bar and send it to someone, they are not guaranteed to see the timestamp you're looking at right now.

To make it clear that you're sharing a link to a specific timestamp, we added the share modal.
- Share modal takes the url that's in the browser url bar, and adds/replaces the `?t=` param with the current timestamp of what you're looking at
- Share modal allows you to override the current timestamp, and type in any time you want, so you can share at a specific timestamp instead.

If the share modal didn't include the `?t=` param then imo it wouldn't really be useful, the user could just go to the browser url bar instead.

----

## Proposal

I think we should tweak the design of the modal, so the checkbox is more like a radio button. Toggling between two options:

An ascii-art mock might be something like:
```
<Title> Share Replay </Title>
<TextInput>https://sentry.sentry.io/replays/javascript:f8491f3ef05b41f89fa106d379b40287/?...snip...&t=6</TextInput>

[ ] Share are current timestamp 12:34
[X] Share at <UserInput> timestamp
```
where the user can choose either A or B but not both

**Comment by jas-kas** [comment]
cc/ @eliashussary

**Comment by ryan953** [comment]
@jas-kas I think this ticket needs to be reworded. 

TL/DR the design of the share modal should be fixed because it's not clear

### Context

The context for this modal is:
- In the base case there's no `?t=` param expected to be in the browser url bar.
- You can actually get `?t=` or `?event_t=` in the browser url bar if you come from pages like Issue Details
  - Either of those params will not update as you play through the video, or click on a console log (for example). So if you copy+paste the url from the browser url bar and send it to someone, they are not guaranteed to see the timestamp you're looking at right now.

To make it clear that you're sharing a link to a specific timestamp, we added the share modal.
- Share modal takes the url that's in the browser url bar, and adds/replaces the `?t=` param with the current timestamp of what you're looking at
- Share modal allows you to override the current timestamp, and type in any time you want, so you can share at a specific timestamp instead.

If the share modal didn't include the `?t=` param then imo it wouldn't really be useful, the user could just go to the browser url bar instead.

----

## Proposal

I think we should tweak the design of the modal, so the checkbox is more like a radio button. Toggling between two options:

An ascii-art mock might be something like:
```
<Title> Share Replay </Title>
<TextInput>https://sentry.sentry.io/replays/javascript:f8491f3ef05b41f89fa106d379b40287/?...snip...&t=6</TextInput>

[ ] Share are current timestamp 12:34
[X] Share at <UserInput> timestamp
```
where the user can choose either A or B but not both

**Comment by Jesse-Box** [comment]
Few questions

1. How would a user share a replay from a specific breadcrumb event? The proposal implies to me that there will be some UI work required on the replay details page to allow users to share from "x" breadcrumb.
2.  How do we handle the case where someone simply wants to share the replay? Radio selection implies to me that the input is based on the radio button selected.
